multiboot2/src/command_line.rs

```diff
@@ -18,6 +28,15 @@ pub struct CommandLineTag {
 }
 
 impl CommandLineTag {
+    /// Create a new command line tag from the given string.
+    #[cfg(feature = "builder")]
+    pub fn new(command_line: &str) -> Box<Self> {
+        let mut bytes: Vec<_> = command_line.bytes().collect();
+        bytes.push(0);
+        let tag = boxed_dst_tag(TagType::Cmdline.into(), &bytes);
+        unsafe { Box::from_raw(Box::into_raw(tag) as *mut Self) }
```

Comment:
`unsafe { Box::from_raw(Box::into_raw(tag) as *mut Self) }`

I think this should be abstracted away in the `boxed_dst_tag` crate, right? Or is there a technical limitation?

Reply:
That was needed because of the [generic casts](https://github.com/rust-osdev/multiboot2/pull/133#issuecomment-1520421043), but it's not anymore with `TagTrait`, you're right!